test/checkpoint/CMakeLists.txt

```diff
@@ -7,52 +7,143 @@ create_test_executable(test_checkpoint
         ${CMAKE_CURRENT_SOURCE_DIR}/recovery-test.sh
 )
 
+# Generates test variants for:
+#   - regular test
+#   - leader test; additional argument: -d leader
+#   - follower test; additional argument -d follower
```

Comment:
is it possible to have disagg test under a different sub directory?
`test/checkpoint`
`test/checkpoint_disagg`